python/lsst/daf/butler/datastores/inMemoryDatastore.py

```diff
@@ -394,17 +439,10 @@ def getUri(self, ref: DatasetRef, predict: bool = False) -> str:
             guessing is not allowed.
 
         """
-
-        # if this has never been written then we have to guess
-        if not self.exists(ref):
-            if not predict:
-                raise FileNotFoundError("Dataset {} not in this datastore".format(ref))
-            name = "{}#predicted".format(ref.datasetType.name)
-        else:
-            realID, _ = self._get_dataset_info(ref)
-            name = '{}'.format(id(self.datasets[realID]))
-
-        return "mem://{}".format(name)
+        uri, _ = self.getURIs(ref, predict)
+        if uri is None:
+            raise RuntimeError(f"Unexpectedly got no URI for in-memory datastore for {ref}")
```

Comment:
This is really an assertion (but I never use assert). I could change it to AssertionError. It's really a "this is impossible" check since getURIs will never do disassembly (so components is false) and None should never be returned if getURIs returns.